alpha/alpha-core/src/main/java/org/apache/servicecomb/saga/alpha/core/EventScanner.java

```diff
@@ -109,6 +113,19 @@ private void updateCompensationStatus(TxEvent event) {
     markSagaEnded(event);
   }
 
+  private void abortTimeoutEvent() {
+    eventRepository.findFirstTimeoutEventByIdGreaterThan(nextTimeoutEventId)
+    .ifPresent((TxEvent event) -> {
+      log.info("Found timeout event {}", event);
+      nextTimeoutEventId = event.id();
+      eventRepository.save(toTxAbortedEvent(event));
```

Comment:
we need to compensate timed out event too, since there's no way to know if its sub tx was done